storage/innobase/xtrabackup/src/ddl_tracker.cc

```diff
@@ -306,16 +336,12 @@ static void copy_for_reduced(copy_thread_ctxt_t *ctxt) {
     if (ctxt->new_tables->find(node->space->id) == ctxt->new_tables->end()) {
       continue;
     }
-    char dest_name[FN_REFLEN];
-    if (fsp_is_file_per_table(node->space->id, node->space->flags)) {
-      snprintf(dest_name, sizeof(dest_name), "%s%s%s",
-               fil_path_to_space_name(node->name), EXT_IBD.c_str(),
-               EXT_NEW.c_str());
-    } else {
-      snprintf(dest_name, sizeof(dest_name), "%s%s",
-               Fil_path::get_basename(node->name).c_str(), EXT_NEW.c_str());
-    }
-    if (xtrabackup_copy_datafile_func(node, num, dest_name)) {
+    // Get destination name for the datafile
+    std::string dest_name = get_dest_name(node);
+    char dest_name_buffer[FN_REFLEN];
+    snprintf(dest_name_buffer, FN_REFLEN, "%s", dest_name.c_str());
+
+    if (xtrabackup_copy_datafile_func(node, num, dest_name_buffer)) {
```

Comment:
simply use dest_name.c_str(). if copy datfile_func signature is char * and not const char*, please modify it